### ISSUE TITLE
Choices are not translated dynamically when using LocaleMiddleware

### ISSUE THREAD

**Issue by EugeneDae** [SmileyChris/django-countries#135]
I've encountered a weird bug with dynamic translations of country names when using `CountryField` in model and `forms.ModelForm`.

**How to reproduce the issue within 60 seconds:**

1. [Download the sample project](http://dae.me/storage/other/django-countries-problem.zip).
2. `pip install django==1.9.2 django-countries`
3. Run the dev server: `python manage.py runserver`
4. Open http://127.0.0.1:8000/fr/ — the request will be passed through `LocaleMiddleware`, which will do, among other things, `translation.activate('fr')`
5. You will see Countries dropdown with French names:
![fr](https://cloud.githubusercontent.com/assets/980835/13128582/d28d6e78-d5e7-11e5-986d-473701d948ee.png)
6. Now, without restarting the server, open the "English" version, http://127.0.0.1:8000/en/ 
7. Everything will be in English, but not the countries dropdown, which will remain in French!

**Interestingly,** if you initialize the field within the form itself, for example:

```
from django_countries.fields import LazyTypedChoiceField
from django_countries import countries

class MyModelForm(forms.ModelForm):
    countries = LazyTypedChoiceField(choices=countries)
    
    class Meta:
        model = MyModel
```

— this will work as expected, and the widget will be picking up different translations just fine.

**To put it in a nutshell: the widget remembers the language it was first rendered in, and keeps it forever until you restart the dev server. This only happens with ModelForms.** 

Also:

1. In the real project I don't actually use `LocaleMiddleware`. I use my own middleware, which simply does `translation.activate(...some_language...)` under certain conditions. So I think `LocaleMiddleware` is not to blame here.

2. I can't reproduce the problem in Django Admin! This could be a clue. You can try it in the sample project: http://127.0.0.1:8000/fr/admin/ and http://127.0.0.1:8000/fr/admin/ (login `test`, password `123hello`)

Thank you!

**Comment by EugeneDae** [comment]
Apparently commenting out the following in django-countries' widgets.py resolves the problem:

```
        if isinstance(self._choices, Promise):
            self._choices = list(self._choices)
```

**Comment by SmileyChris** [comment]
I think you'll find this fixed in 4.0. A change in Django 1.9 introduced this bug (deepcopy method on widgets was caching the widget choices)
